### PR TITLE
OCPBUGS#12214: Add example for bond+vlan

### PR DESCRIPTION
**Pull request by snarayan-redhat**
Version(s): 4.12+

Issue: https://issues.redhat.com/browse/OCPBUGS-12214

Link to docs preview: https://59345--docspreview.netlify.app/openshift-enterprise/latest/networking/k8s_nmstate/k8s-nmstate-updating-node-network-config.html#virt-example-nmstate-multiple-interfaces_k8s_nmstate-updating-node-network-config

QE review:
- [x] QE has approved this change. @rbbratta
- [x] SME has approved this change. @cybertron

### CODE REVIEW

File: modules/virt-example-nmstate-multiple-interfaces.adoc [hunk 1]
```diff
@@ -7,26 +7,40 @@
...
+kind: NodeNetworkConfigurationPolicy
+metadata:
+  name: bond-vlan
+spec:
+  nodeSelector:
+    kubernetes.io/hostname: dell-per740-14.rhts.eng.pek2.redhat.com
+  desiredState:
     interfaces:
     - name: bond10
       description: Bonding eth2 and eth3 for Linux bridge
```
Comment (cybertron) [inline modules/virt-example-nmstate-multiple-interfaces.adoc:24]: Description needs to be updated.

File: modules/virt-example-nmstate-multiple-interfaces.adoc [hunk 2]
```diff
@@ -7,26 +7,40 @@
...
-      bridge:
-        port:
-        - name: bond10
-...
+      vlan:
+         base-iface: bond10
+         id: 103
+      ipv4:
+        address:
+        - ip: 33.103.3.10
```
Comment (cybertron) [inline modules/virt-example-nmstate-multiple-interfaces.adoc:43]: Maybe just use DHCP so we don't have to use specific IPs. In that case, this section would look like:
```
ipv4:
  dhcp: true
  enabled: true
```

**Comment by cybertron** [comment]
/lgtm

Thanks!

File: modules/virt-example-nmstate-multiple-interfaces.adoc [hunk 3]
```diff
@@ -7,26 +7,53 @@
...
-      type: bond
-      state: up
+    - name: bond10 <4>
+      description: Bonding eth2 and eth3 <5>
+      type: bond <6>
+      state: up <7>
       link-aggregation:
-        port:
+        mode: balance-rr <8>
+         options:
```
Comment (rbbratta) [inline modules/virt-example-nmstate-multiple-interfaces.adoc:29]: bad YAML?
 `options:` should be same indent as `mode:`?

Comment (snarayan-redhat) [reply modules/virt-example-nmstate-multiple-interfaces.adoc:29]: yes, that's right. Made the correction.

File: modules/virt-example-nmstate-multiple-interfaces.adoc [hunk 4]
```diff
@@ -7,26 +7,53 @@
...
-      description: Bonding eth2 and eth3 for Linux bridge
-      type: bond
-      state: up
+    - name: bond10 <4>
+      description: Bonding eth2 and eth3 <5>
+      type: bond <6>
+      state: up <7>
       link-aggregation:
-        port:
+        mode: balance-rr <8>
```
[the remark this answers is not in the document]
Comment (rbbratta) [reply modules/virt-example-nmstate-multiple-interfaces.adoc:28]: Correct, no `"` needed.  It was a false-positive syntax highlight in vs.code.

[the remark this answers is not in the document]
Comment (snarayan-redhat) [reply modules/virt-example-nmstate-multiple-interfaces.adoc:28]: if everything else looks good, can you provide an lgtm?

File: modules/virt-example-nmstate-multiple-interfaces.adoc [hunk 5]
```diff
@@ -7,26 +7,53 @@
...
+        options:
+         miimon: '140' <9>
```
Comment (rbbratta) [inline modules/virt-example-nmstate-multiple-interfaces.adoc:30]: micro-nit, I think we use two spaces for indent. `miimon:` is one space indented.

Comment (snarayan-redhat) [reply modules/virt-example-nmstate-multiple-interfaces.adoc]: Thanks for catching that!

File: modules/virt-example-nmstate-multiple-interfaces.adoc [hunk 6]
```diff
@@ -7,26 +7,53 @@
 
 You can create multiple interfaces in the same node network configuration policy. These interfaces can reference each other, allowing you to build and deploy a network configuration by using a single policy manifest.
 
-The following example snippet creates a bond that is named `bond10` across two NICs and a Linux bridge that is named `br1` that connects to the bond.
+The following example snippet creates a bond that is named `bond10` across two NICs and VLAN that is named `bond10.103` that connects to the bond.
```
Comment (GroceryBoyJr) [inline modules/virt-example-nmstate-multiple-interfaces.adoc:10]:
```suggestion
The following example YAML file creates a bond that is named `bond10` across two NICs and VLAN that is named `bond10.103` that connects to the bond.
```

Comment (GroceryBoyJr) [reply modules/virt-example-nmstate-multiple-interfaces.adoc:10]: @snarayan-redhat changing from "snippet" to "YAML file" eliminates ambiguity and continues the term used in the prior YAML example.

**Review by GroceryBoyJr — COMMENTED** [review]
@snarayan-redhat one small nit, otherwise LGTM, good work!

**Pull request merged**